src/bn254/validation.rs

```diff
@@ -0,0 +1,143 @@
+use arrayvec::ArrayVec;
+use boojum::cs::traits::cs::ConstraintSystem;
+use boojum::field::SmallField;
+use boojum::gadgets::boolean::Boolean;
+use boojum::gadgets::non_native_field::implementations::NonNativeFieldOverU16Params;
+use boojum::gadgets::non_native_field::traits::NonNativeField;
+use boojum::gadgets::u256::UInt256;
+use std::sync::Arc;
+
+use crate::bn254::{BN256BaseNNField, BN256BaseNNFieldParams, BN256Fq, BN256Fq2NNField};
+use crate::ethereum_types::U256;
+use boojum::pairing::ff::PrimeField;
+
+// The Short Weierstrass equation of the curve is  y^2 = x^3 + B.
+// B parameter for BN256 curve equation.
+const B: &str = "3";
+const B_TWIST_C0: &str =
+    "19485874751759354771024239261021720505790618469301721065564631296452457478373";
+const B_TWIST_C1: &str =
+    "266929791119991161246907387137283842545076965332900288569378510910307636690";
+
```

Comment:
if they dont exist already, it would be helpful to include tests for all the functions in this file just to help test coverage and have a more robust pipeline